src/utils/salesEvidenceFilter.ts

```diff
@@ -0,0 +1,47 @@
+import { SalesEvidenceFilterParams, SaleTypeEnum, UnprocessedResultsFromCRM, MinMaxNumberType, MinMaxDateType } from '../types'
+
+function numberFilter (filterType: string, filterValues: MinMaxNumberType, property: UnprocessedResultsFromCRM) {
+    return typeof property[filterType] === 'number' && property[filterType] >= filterValues.min && property[filterType] <= filterValues.max
+}
+
+function dateFilter (dateSold: MinMaxDateType, property: UnprocessedResultsFromCRM): boolean {
+    if (typeof dateSold.min !== 'undefined' && typeof dateSold.max !== 'undefined') {
+        const minDate = dateSold.min.toISOString().split('T')[0]
+        const maxDate = dateSold.max.toISOString().split('T')[0]
+        return !property.Sale_Date && property.Sale_Date >= minDate && property.Sale_Date <= maxDate
```

Comment:
I don't understand this. Seems like this would never evaluate to true. ```!property.Sale_Date && property.Sale_Date >= minDate```

Should it be
```
property.Sale_Date && property.Sale_Date >= minDate
```

Reply:
I'm going to change it to `!!property.Sale_Date` b/c i'll get this type error: `Type 'boolean | ""' is not assignable to type 'boolean'. Type '""' is not assignable to type 'boolean'.ts(2322)`.